### PR TITLE
Add dry run for loaders to check for valid instances

### PR DESCRIPTION
**Pull request by anwai98**
Resolves https://github.com/computational-cell-analytics/micro-sam/issues/703

### CODE REVIEW

File: micro_sam/training/training.py [hunk 1]
```diff
@@ -71,38 +73,44 @@ def check_instance_channel(instance_channel):
...
-    else:
-        if n_channels_y != 1:
-            raise ValueError(
-                "Invalid number of channels in the target data from the data loader. "
-                "Expect 1 channel for training without an instance segmentation decoder,"
-                f"but got {n_channels_y} channels."
-            )
-        check_instance_channel(y)
+    name = "" if name is None else f"'{name}'"
+    for x, y in tqdm(loader, desc=f"Verifying labels in {name} dataloader"):
```
Comment (constantinpape) [inline micro_sam/training/training.py:77]: I am a bit hesitant to do this by default. If we have a large data loader it may delay the start of training quite a bit. Let's discuss on Monday. I think the best solution would be to define a number of samples that will be checked to make this flexible.

**Comment by constantinpape** [comment]
Fyi @anwai98 I changed the target branch to `dev`, since this contained all recent `dev` commits.

File: micro_sam/training/training.py (hunk 1 above)
Comment (anwai98) [reply micro_sam/training/training.py]: I exposed an additional arguments which controls whether we check labels for the entire dataloader or for the first n samples. I have left the current defaults for it to `None`, that corresponds to checking the entire dataloader.

File: micro_sam/training/training.py [hunk 2]
```diff
@@ -157,6 +174,7 @@ def train_sam(
     pbar_signals: Optional[QObject] = None,
     optimizer_class: Optional[Optimizer] = torch.optim.AdamW,
     peft_kwargs: Optional[Dict] = None,
+    verify_n_labels_in_loader: Optional[int] = None,
```
Comment (constantinpape) [inline micro_sam/training/training.py:177]: I would set the default to something rather small (maybe 50). Otherwise `None` causes quite long wait times for a big data loader.

Comment (anwai98) [reply micro_sam/training/training.py]: Yep, that sounds reasonable. I've updated the defaults now to `50`. Thanks!

**Comment by anwai98** [comment]
Yayy, the windows CI works now! 🥳 

cc: @constantinpape

**Review by constantinpape — APPROVED** [review]
